libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFPushNotificationManager.m

```diff
@@ -182,7 +183,14 @@ - (BOOL)registerSalesforceNotificationsWithCompletionBlock:(SFUserAccount *)user
             [[SFPreferences currentUserLevelPreferences] setObject:strongSelf->_deviceSalesforceId forKey:kSFDeviceSalesforceId];
             [[SFPreferences currentUserLevelPreferences] synchronize];
             [SFSDKCoreLogger i:[strongSelf class] format:@"Response:%@", responseAsJson];
-            [strongSelf postPushNotificationRegistration:completionBlock];
+            [strongSelf getNotificationTypesWithRestClient:[SFRestAPI sharedInstance]
+                                              userDefaults:[NSUserDefaults standardUserDefaults]
+                                         completionHandler:^(BOOL passed, NSError * _Nullable error) {
+                if (error != nil) {
+                    [SFSDKCoreLogger e:[strongSelf class] format:@"Get Notification Types Error: %@", [error localizedDescription]];
+                }
+                [strongSelf postPushNotificationRegistration:completionBlock];
+            }];
```

Comment:
For Registration and Re-registration to get the notification types.   Notice we call the post registration after we get the notification types.  @wmathurin can you confirm this the expected behavior?

Reply:
I think so.